lib/ansible/template/__init__.py

```diff
@@ -996,7 +996,7 @@ def _lookup(self, name, *args, **kwargs):
                 return wrap_var(ran)
 
             try:
-                if isinstance(ran[0], NativeJinjaText):
+                if isinstance(list(ran)[0], NativeJinjaText):
```

Comment:
or simpler above `if wantlist or not isinstance(ran, list)`

Reply:
ah, true, then the 'fix' would be to add KeyError, IndexError to TypeError catch

Reply:
fyi, lookup plugins were not required to return lists, this is something that we decided much later on and never enforced at the lookup plugin, specially with the `lookup` form, the `q` or `query` form of invocation does enforce this

Reply:
@mkrizek that is correct. I guess ideal behavior would be returning the dictionary, but expected behavior would be a comma-separated string of keys. I'll revert to not introduce any breaking changes.

Reply:
> ah, true, then the 'fix' would be to add KeyError, IndexError to TypeError catch

In that case, can we get rid of the `AnsibleError` being raised?

Reply:
No, it is fine, it just covers more errors when 'not a list'

Reply:
Got it. Pushed a fix let me know if it needs any changes

Reply:
test/integration/targets/templating_lookups/